apps/server/lib/lexical/server/transport/std_io.ex

```diff
@@ -76,29 +83,62 @@ defmodule Lexical.Server.Transport.StdIO do
     end
   end
 
-  defp parse_headers(headers) do
-    Enum.map(headers, &parse_header/1)
-  end
+  defp content_length(headers) do
+    case List.keyfind(headers, "content-length", 0) do
+      {_, len_str} ->
+        {:ok, String.to_integer(len_str)}
 
-  defp header_value(headers, header_name, converter) do
-    case List.keyfind(headers, header_name, 0) do
-      nil -> :error
-      {_, value} -> {:ok, converter.(value)}
+      nil ->
+        {:error, :no_content_length}
     end
   end
 
   defp read_body(device, byte_count) do
     case IO.binread(device, byte_count) do
-      data when is_binary(data) or is_list(data) ->
+      data when is_binary(data) and byte_size(data) === byte_count ->
         {:ok, data}
 
-      other ->
-        other
+      data when is_binary(data) ->
+        read_more(device, byte_count - byte_size(data), data)
+
+      data when is_list(data) ->
+        if Enum.count(data) === byte_count do
```

Comment:
```suggestion
        if Enum.count(data) == byte_count do
```
javascript contamination

Reply:
fixed